ops-catalog/stats.schema.yaml

```diff
@@ -0,0 +1,157 @@
+    reduce:
+      strategy: sum
+  txnCount:
+    description: Total number of transactions represented by this stats document
+    type: integer
+    reduce:
+      strategy: sum
+  capture:
+    description:
+      Capture stats, organized by collection. The keys of this object are
+      the collection names, and the values are the stats for that collection.
+    type: object
+    additionalProperties:
+      type: object
+      properties:
+        right:
+          description: Documents fed into the combiner from the source
+          $ref: "#/$defs/docsAndBytes"
+        out:
+          $ref: "#/$defs/docsAndBytes"
+      reduce:
+        strategy: merge
+    reduce:
+      strategy: merge
+  materialize:
+    description:
+      A map of each binding source (collection name) to combiner stats
+      for that binding
+    type: object
+    additionalProperties:
+      type: object
+      properties:
+        left:
+          $ref: "#/$defs/docsAndBytes"
+        right:
+          $ref: "#/$defs/docsAndBytes"
+        out:
+          type: object
+          properties:
+            docsTotal:
+              description: Total number of documents
+              type: integer
+              default: 0
+              reduce:
+                strategy: sum
+            bytesTotal:
+              description: Total number of bytes representing the JSON encoded documents
+              type: integer
+              default: 0
+              reduce:
+                strategy: sum
+          reduce:
+            strategy: merge
+          required:
+            - docsTotal
+      reduce:
+        strategy: merge
+    reduce:
+      strategy: merge
+  derive:
+    type: object
+    properties:
+      transforms:
+        description:
+          A map of each transform (transform name, not collection name)
+          to stats for that transform
+        type: object
+        additionalProperties:
+          $ref: "#/$defs/transformStats"
+        reduce:
+          strategy: merge
+      published:
+        $ref: "#/$defs/docsAndBytes"
+      out:
+        $ref: "#/$defs/docsAndBytes"
+    reduce:
+      strategy: merge
+  interval:
+    type: object
+    properties:
+      uptimeSeconds:
+        description:
+          Number of seconds that the task shard is metered as having been
+          running
+        type: integer
+        reduce:
+          strategy: sum
+        minimum: 1
+      usageRate:
+        description:
+          Usage rate which adjusts `uptimeSeconds` to determine the task's
+          effective usage
+        type: number
+        minimum: 0
+        default: 0
+    required:
+      - uptimeSeconds
+    reduce:
+      strategy: merge
+reduce:
+  strategy: merge
+required:
+  - shard
+  - ts
+$defs:
+  docsAndBytes:
+    type: object
+    properties:
+      docsTotal:
+        description: Total number of documents
+        type: integer
+        default: 0
+        reduce:
+          strategy: sum
+      bytesTotal:
```

Comment:
`bytesTotal` seems like it's still not nullable, so not sure why it was in the TS code.

Stopping here for now. Looks great so far

Reply:
it's not nullable, but it's also not marked as required. Dunno when we did that, but this apparently hasn't been re-generated since.

I didn't edit this yaml -- I merely ran it through a JSON => yaml converter to make everything consistently YAML and tweaked file names, without making any content changes. Which I now kinda wish I hadn't done, but ... :shrug: